testsuite/lb_poiseuille_4.0.py

```diff
@@ -0,0 +1,178 @@
+
+        mid_indices = [int((self.system.box_l[0] / AGRID) / 2),
+                       int((self.system.box_l[1] / AGRID) / 2),
+                       int((self.system.box_l[2] / AGRID) / 2)]
+        diff = float("inf")
+        old_val = self.lbf[mid_indices].velocity[2]
+        while diff > 0.005:
+            self.system.integrator.run(100)
+            new_val = self.lbf[mid_indices].velocity[2]
+            diff = abs(new_val - old_val)
+            old_val = new_val
+
+    def test_profile(self):
+        """
+        Compare against analytical function by calculating the RMSD.
+
+        """
+        self.prepare()
+        velocities = np.zeros((int(self.system.box_l[0] / AGRID), 2))
+
+        for x in range(velocities.shape[0]):
+            v_tmp = []
+            for y in range(int(self.system.box_l[1] + 1)):
+                for z in range(int(self.system.box_l[2] + 1)):
+                    v_tmp.append(self.lbf[x, y, z].velocity[2])
+            velocities[x, 1] = np.mean(np.array(v_tmp))
+            velocities[x, 0] = (x + 0.5) * AGRID
+
+        v_measured = velocities[1:-1, 1]
+        v_expected = poiseuille_flow(velocities[1:-1,
+                                                0] - 0.5 * self.system.box_l[
+                                                    0],
+                                     self.system.box_l[0] - 2.0 * AGRID,
+                                     EXT_FORCE,
+                                     VISC * DENS)
+        rmsd = np.sqrt(np.sum(np.square(v_expected - v_measured)))
+        self.assertLess(rmsd, 0.02 * AGRID / TIME_STEP)
+
+
+@ut.skipIf(not espressomd.has_features(
+    ['LB_GPU', 'LB_BOUNDARIES_GPU', 'EXTERNAL_FORCES']), "Skipping test due to missing features.")
+class LBGPUPoiseuille(ut.TestCase, LBPoiseuilleCommon):
+
+    """Test for the GPU implementation of the LB."""
+
+    def setUp(self):
+        self.lbf = espressomd.lb.LBFluidGPU(**LB_PARAMS)
+
+
+@ut.skipIf(not espressomd.has_features(
+    ['LB_GPU', 'LB_BOUNDARIES_GPU', 'EXTERNAL_FORCES']), "Skipping test due to missing features.")
+class LBGPUPoiseuilleInterpolation(ut.TestCase, LBPoiseuilleCommon):
+
+    """Test for the higher order interpolation scheme of the LB."""
+
+    def setUp(self):
+        self.lbf = espressomd.lb.LBFluidGPU(**LB_PARAMS)
+
+    def test_profile(self):
+        """
+        Compare against analytical function by calculating the RMSD.
+
+        """
+        self.prepare()
+        velocities = np.zeros((50, 2))
+        x_values = np.linspace(
+            2.0 * AGRID,
+            self.system.box_l[0] - 2.0 * AGRID,
+            50)
+
+        cnt = 0
+        for x in x_values:
+            v_tmp = []
+            for y in range(int(self.system.box_l[1] + 1)):
+                for z in range(int(self.system.box_l[2] + 1)):
+                    v_tmp.append(
+                        self.lbf.get_interpolated_fluid_velocity_at_positions(np.array([[x, y * AGRID, z * AGRID],]))[0][2]
```

Comment:
the same test can be run on the current python branch by replacing this line with
``` python
self.lbf.get_interpolated_fluid_velocity_at_positions(np.array([[x, y * AGRID, z * AGRID],]), True)[0][2]
```